### PR TITLE
Allow relative urls for fastboot and ember-data

### PR DESCRIPTION
**Pull request by rwwagner90**
Copied over the `buildServerUrls` function from ember-fetch-adapter to allow relative urls with fastboot and ember-data.

### CODE REVIEW

File: addon/mixins/adapter-fetch.js [hunk 1]
```diff
@@ -165,7 +169,20 @@ export function determineBodyPromise(response, requestData) {
 }
 
 export default Mixin.create({
-  adapter: service(),
+  fastboot: computed(function() {
+    let owner = getOwner(this);
+    return owner && owner.lookup('service:fastboot');
+  }),
+
+  protocol: computed(function() {
```
Comment (nlfurniss) [inline addon/mixins/adapter-fetch.js]: why use CPs with dependent keys? Can these be set in `init`?

**Comment by rwwagner90** [comment]
@nlfurniss I copied this stuff from https://github.com/tchak/ember-fetch-adapter/blob/master/addon/index.js

It allows users of this mixin to make requests when in fastboot. Without these changes, you get an error with node-fetch complaining about `only absolute URLs are supported`.

**Comment by rwwagner90** [comment]
I have been speaking with @tchak about this. Might be good for him to weigh in here 😃

**Comment by rwwagner90** [comment]
@nlfurniss @tchak any updates on this?

**Comment by chrism** [comment]
I've been getting this error too when using with Fastboot after following the instructions.

`Error while processing route: index Only absolute URLs are supported TypeError: Only absolute URLs are supported`

It would be great to see this merged.

**Comment by rwwagner90** [comment]
@nlfurniss @rwjblue what do we need to do to get this merged?

**Comment by nlfurniss** [comment]
Can you rebase and resolve the conflicts? Also what about tests?

**Comment by rwwagner90** [comment]
@nlfurniss yes, I can rebase, but tests would need a very special setup to force fastboot and ember-data, and I am not sure it would be something we could easily do. If I rebase, can we merge as is? This has been open quite awhile.

File: addon/mixins/adapter-fetch.js [hunk 2]
```diff
@@ -251,13 +271,38 @@ export default Mixin.create({
...
- * Allows for the error to be selected from either the
- * response object, or the response data.
- * @param {Object} response
- * @param {Object} payload
- */
+    if (protocolRelativeRegex.test(url)) {
+      return `${protocol}${url}`;
+    } else if (!httpRegex.test(url)) {
+      try {
+        return `${protocol}//${host}${url}`;
```
Comment (xg-wang) [inline addon/mixins/adapter-fetch.js]: When will this throw?

**Comment by rwwagner90** [comment]
> I copied this stuff from https://github.com/tchak/ember-fetch-adapter/blob/master/addon/index.js
> 
> It allows users of this mixin to make requests when in fastboot. Without these changes, you get an error with node-fetch complaining about `only absolute URLs are supported`.

@xg-wang I just copied all this, as mentioned previously. Since I know it works currently, and it's copied from this other addon, I am not sure what the best way to proceed is.

File: addon/mixins/adapter-fetch.js (hunk 1 above)
Comment (xg-wang) [reply addon/mixins/adapter-fetch.js]: @chrism Yea you're absolutely right, thansk for pointing out!

**Comment by xg-wang** [comment]
Close by https://github.com/ember-cli/ember-fetch/pull/143

**Pull request closed without merging**